ambry-store/src/test/java/com.github.ambry.store/StatsManagerTest.java

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.Partition;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.PartitionState;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.StatsManagerConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Utils;
+import com.github.ambry.utils.UtilsTest;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests for {@link StatsManager}.
+ */
+public class StatsManagerTest {
+  private final StatsManager statsManager;
+  private final Map<PartitionId, Store> storeMap;
+  private final String outputFileString;
+  private final File tempDir;
+  private final List<StatsSnapshot> statsSnapshots;
+  private final StatsSnapshot preAggregatedSnapshot;
+  private final Random random = new Random();
+  private final StatsManagerConfig config;
+
+  /**
+   * Deletes the temporary directory.
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @After
+  public void cleanup() throws IOException {
+    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+  }
+
+  public StatsManagerTest() throws IOException, StoreException {
+    tempDir = StoreTestUtils.createTempDirectory("nodeStatsDir-" + UtilsTest.getRandomString(10));
+    outputFileString = (new File(tempDir.getAbsolutePath(), "stats_output.json")).getAbsolutePath();
+    int[][] testData1 = new int[][]{{1100, 250, 600}, {330}, {500, 300}, {1000, 2000, 3000, 50, 150}};
+    preAggregatedSnapshot = generateStatsSnapshot(testData1);
+    int[][] testData2 = decomposeArray(testData1);
+    int[][] testData3 = decomposeArray(testData1);
+    storeMap = new HashMap<>();
+    statsSnapshots = new ArrayList<>();
```

Comment:
btw, you don't need this dataStr at all. line 80 : Store store = new MockBlobStore(new MockBlobStoreStats(generateStatsSnapshot(testData1), false));